daemon/cmd/policy.go

```diff
@@ -36,12 +37,51 @@ import (
 	"github.com/cilium/cilium/pkg/option"
 	"github.com/cilium/cilium/pkg/policy"
 	policyAPI "github.com/cilium/cilium/pkg/policy/api"
+	"github.com/cilium/cilium/pkg/promise"
 	"github.com/cilium/cilium/pkg/safetime"
 	"github.com/cilium/cilium/pkg/trigger"
 )
 
+func newPolicyRepository(
+	identityAllocatorPromise promise.Promise[cache.IdentityAllocator],
+	epMgr endpointmanager.EndpointManager,
+	certManager certificatemanager.CertificateManager,
+	secretManager certificatemanager.SecretManager,
+) (promise.Promise[*policy.Repository], promise.Promise[*policy.Updater]) {
+	repoPromise := promise.Map(identityAllocatorPromise, func(alloc cache.IdentityAllocator) *policy.Repository {
+		repo := policy.NewPolicyRepository(alloc,
+			alloc.GetIdentityCache(),
+			certManager,
+			secretManager,
+		)
+		repo.SetEnvoyRulesFunc(envoy.GetEnvoyHTTPRules)
+		return repo
+	})
+
+	updaterResolver, updaterPromise := promise.New[*policy.Updater]()
```

Comment:
You can use `promise.Map` here as well. Prefer `Map()` if you can as then the caller doing the Await() will provide the context and we don't need to conjure it here.